extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -79,8 +120,17 @@ void validateBeanDeployment(
             if (bean.isClassBean()) {
                 // TODO: add support for inherited business methods
```

Comment:
The `AnnotationsTransformer` above does support inherited business methods but we ignore these methods here. I think we should be consistent and improve this part as well ;-).

Reply:
Otherwise LGTM.

Reply:
Hm I'm looking at the spec and its API, and AFAICS the `@Incoming` and `@Outgoing` annotations are not inherited, so I guess we don't need to care about inheritance here.
In that case it would rather make sense to remove the inheritance tree walking in `AnnotationsTransformer`? Is that right?

Reply:
Well, it seems the spec does not define this. So I would take into account the behavior defined by Commons Annotations: _"Members inherited from a superclass and which are not hidden or overridden maintain the annotations they had in the class that declared them..."_. In other words, those methods should be probably inherited.

However, the algorithm wouldn't be that simple, for example if we want to handle generics correctly.

And so I'd suggest to remove the inheritance processing from the annotation transformer and file a new issue to address the inheritance problem properly.

Reply:
Ok, I removed the inheritance processing  and filed https://github.com/quarkusio/quarkus/issues/4106